vertical-pod-autoscaler/pkg/recommender/routines/cpu_integer_post_processor_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routines
+
+import (
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/model"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/test"
+	"testing"
+)
+
+func Test_extractContainerName(t *testing.T) {
+	tests := []struct {
+		name   string
+		key    string
+		prefix string
+		suffix string
+		want   string
+	}{
+		{
+			name:   "empty",
+			key:    "",
+			prefix: "",
+			suffix: "",
+			want:   "",
+		},
+		{
+			name:   "no match",
+			key:    "abc",
+			prefix: "z",
+			suffix: "x",
+			want:   "",
+		},
+		{
+			name:   "match",
+			key:    "abc",
+			prefix: "a",
+			suffix: "c",
+			want:   "b",
+		},
+		{
+			name:   "real",
+			key:    vpaPostProcessorPrefix + "kafka" + vpaPostProcessorIntegerCPUSuffix,
+			prefix: vpaPostProcessorPrefix,
+			suffix: vpaPostProcessorIntegerCPUSuffix,
+			want:   "kafka",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, extractContainerName(tt.key, tt.prefix, tt.suffix), "extractContainerName(%v, %v, %v)", tt.key, tt.prefix, tt.suffix)
+		})
+	}
+}
+
+func Test_integerCPUPostProcessor_Process(t *testing.T) {
+	tests := []struct {
+		name           string
+		vpa            *model.Vpa
+		recommendation *vpa_types.RecommendedPodResources
+		want           *vpa_types.RecommendedPodResources
+	}{
+		{
+			name: "No containers match",
+			vpa: &model.Vpa{Annotations: map[string]string{
+				vpaPostProcessorPrefix + "container-other" + vpaPostProcessorIntegerCPUSuffix: "true",
```

Comment:
Please replace `"true"` with `vpaPostProcessorIntegerCPUValue` here and below

Reply:
done